### PR TITLE
Report missing inputs in sendrawtransaction

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

**Comment by EagleTM** [comment]
I've applied this patch to 0.10.0 for debugging purposes and it works well and detects lots of {u'message': u'Missing inputs', u'code': -26} on a public electrum server. I believe these were reported as code -25 before applying the patch, because this is the only error code I can find in logs which does not have a message. Moreover "Missing inputs" accounts for 70% of u'code': -26 messages and the overall rate has gone up, while code -25 decreased.

**Comment by sipa** [comment]
Rebased and cherry-picked @jonasschnelli's RPC test.

**Comment by laanwj** [comment]
utACK, needs rebase though

File: src/rpcrawtransaction.cpp [hunk 1]
```diff
@@ -741,11 +741,16 @@ Value sendrawtransaction(const Array& params, bool fHelp)
...
         CValidationState state;
-        if (!AcceptToMemoryPool(mempool, state, tx, false, NULL, !fOverrideFees)) {
+        bool fMissingInputs;
+        if (!AcceptToMemoryPool(mempool, state, tx, false, &fMissingInputs, !fOverrideFees)) {
             if(state.IsInvalid())
                 throw JSONRPCError(RPC_TRANSACTION_REJECTED, strprintf("%i: %s", state.GetRejectCode(), state.GetRejectReason()));
-            else
+            else {
+                if (fMissingInputs) {
+                    throw JSONRPCError(RPC_TRANSACTION_REJECTED, "Missing inputs");
```
Comment (sipa) [inline src/rpcrawtransaction.cpp]: Changed to RPC_TRANSACTION_ERROR.

**Comment by sipa** [comment]
Rebased and nit addressed.

**Comment by laanwj** [comment]
Fails travis, the test needs to be updated to use `generate`:
```
Running testscript rawtransactions.py...
Initializing test directory /tmp/testtgQ972
JSONRPC error: Use the generate method instead of setgenerate on this network
File "/home/travis/build/bitcoin/bitcoin/bitcoin-i686-pc-linux-gnu/qa/rpc-tests/test_framework.py", line 119, in main
self.run_test()
File "/home/travis/build/bitcoin/bitcoin/bitcoin-i686-pc-linux-gnu/qa/rpc-tests/rawtransactions.py", line 42, in run_test
self.nodes[2].setgenerate(True, 1)
File "/home/travis/build/bitcoin/bitcoin/bitcoin-i686-pc-linux-gnu/qa/rpc-tests/python-bitcoinrpc/bitcoinrpc/authproxy.py", line 126, in __call__
raise JSONRPCException(response['error'])
Stopping nodes
Cleaning up
iled
```

**Comment by jonasschnelli** [comment]
New test is here: https://github.com/jonasschnelli/bitcoin/commit/70eccdf6e29f83de4f239107bbf9d21ee44401b5

**Comment by sipa** [comment]
Updated.

**Comment by land-pack** [comment]
It's fixed?

**Pull request merged**